common/include/opae/cxx/handle.h

```diff
@@ -36,14 +36,26 @@ namespace opae {
 namespace fpga {
 namespace types {
 
+/** An allocated accelerator resource
+ *
+ * Represents an accelerator resource that has
+ * been allocated by OPAE. The accelerator's
+ * MMIO regions are made available for reading
```

Comment:
Hmm. MMIO is implementation for register access. If you want to keep things generic, maybe you could rephrase this part:

>Depending on the type of resource, its registers space may be read/written using a handle object.

  We may also think about this too for the documentation for the read/write functions.